python/pyarrow/_dataset.pyx

```diff
@@ -733,30 +721,120 @@ cdef class Dataset:
     cdef inline shared_ptr[CDataset] unwrap(self) nogil:
         return self.wrapped
 
-    def new_scan(self, MemoryPool memory_pool=None):
+    def scan(self, columns=None, filter=None, MemoryPool memory_pool=None):
+        """Builds a scan operation against the dataset.
+
+        It poduces a stream of ScanTasks which is meant to be a unit of work to
+        be dispatched. The tasks are not executed automatically, the user is
+        responsible to execute and dispatch the individual tasks, so custom
+        local task scheduling can be implemented.
+
+        Parameters
+        ----------
+        columns : list of str, default None
+            List of columns to project. Order and duplicates will be preserved.
+            The columns will be passed down to Sources and corresponding data
+            fragments to avoid loading, copying, and deserializing columns
+            that will not be required further down the compute chain.
+            By default all of the available columns are projected. Raises
+            an exception if any of the referenced column names does not exist
+            in the dataset's Schema.
+        filter : Expression, default None
+            Scan will return only the rows matching the filter.
+            If possible the predicate will be pushed down to exploit the
+            partition information or internal metadata found in the data
+            source, e.g. Parquet statistics. Otherwise filters the loaded
+            RecordBatches before yielding them.
+        memory_pool : MemoryPool, default None
+            For memory allocations, if required. If not specified, uses the
+            default pool.
+
+        Returns
+        -------
+        scan_tasks : iterator of ScanTask
         """
-        Begin to build a new Scan operation against this Dataset.
+        scanner = Scanner(self, columns=columns, filter=filter,
+                          memory_pool=memory_pool)
+        return scanner.scan()
```

Comment:
An option is also to return the `scanner` object itself?

Reply:
The scanner object can be easily instantiated by `Scanner(dataset, **kwargs)`